### PR TITLE
WIP - Rename dumbblobstore to blobstore

### PR DESCRIPTION
**Pull request by remk**

### CODE REVIEW

File: server/blob/blob-api/pom.xml [hunk 1]
```diff
@@ -54,7 +54,6 @@
         <dependency>
             <groupId>commons-io</groupId>
             <artifactId>commons-io</artifactId>
-            <scope>test</scope>
```
Comment (chibenwa) [inline server/blob/blob-api/pom.xml]: Why?

File: server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java [hunk 2]
```diff
@@ -291,7 +291,7 @@ void constructorShouldNotRegisterGetQueueSizeGaugeWhenSizeMetricsDisabled(MailQu
     }
 
     private void setUp(CassandraCluster cassandra, MailQueueMetricExtension.MailQueueMetricTestSystem metricTestSystem, RabbitMQMailQueueConfiguration configuration) throws Exception {
-        CassandraDeduplicatingBlobStore blobStore = CassandraDeduplicatingBlobStore.forTesting(cassandra.getConf());
+        DeduplicatingBlobStore blobStore = CassandraBlobStoreFixture.storeForTesting(cassandra.getConf());
```
Comment (chibenwa) [inline server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java]: Can't we have a similar fixture for memory?

File: server/blob/blob-api/pom.xml (hunk 1 above)
Comment (remk) [reply server/blob/blob-api/pom.xml]: We use it in `DeduplicatingBlobStoreImpl`

**Comment by remk** [comment]
> Why don't we use `DeduplicatingBlobStoreImpl` With object storage?

This PR is a work in progress at the moment. It is not yet rebased on the object storage "dumb blobstore" PRs.

**Comment by remk** [comment]
```
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1) No implementation for org.apache.james.blob.api.BlobStore was bound.
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   while locating org.apache.james.blob.api.BlobStore
[b5e8b12158a8a419f20041ff3cc374de36640fc9]     for the 2nd parameter of org.apache.james.blob.api.DeduplicatingBlobStoreImpl.<init>(DeduplicatingBlobStoreImpl.java:43)
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   at org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils.lambda$commonModules$2(GuiceUtils.java:75) (via modules: com.google.inject.util.Modules$CombinedModule -> org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils$$Lambda$548/0x00000001005ff040)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1 error
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 	at org.apache.james.mailbox.cassandra.mail.CassandraMessageIdMapperTest.provideMapper(CassandraMessageIdMapperTest.java:50)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 	at org.apache.james.mailbox.cassandra.mail.CassandraMessageIdMapperTest.provideMapper(CassandraMessageIdMapperTest.java:42)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] [ERROR] deleteWithMailboxIdsShouldDeleteIndicesWhenMailboxIdsContainsMultipleElements  Time elapsed: 0.015 s  <<< ERROR!
[b5e8b12158a8a419f20041ff3cc374de36640fc9] com.google.inject.CreationException: 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] Unable to create injector, see the following errors:
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1) No implementation for org.apache.james.blob.api.BlobStore was bound.
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   while locating org.apache.james.blob.api.BlobStore
[b5e8b12158a8a419f20041ff3cc374de36640fc9]     for the 2nd parameter of org.apache.james.blob.api.DeduplicatingBlobStoreImpl.<init>(DeduplicatingBlobStoreImpl.java:43)
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   at org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils.lambda$commonModules$2(GuiceUtils.java:75) (via modules: com.google.inject.util.Modules$CombinedModule -> org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils$$Lambda$548/0x00000001005ff040)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1 error
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 	at org.apache.james.mailbox.cassandra.mail.CassandraMessageIdMapperTest.provideMapper(CassandraMessageIdMapperTest.java:50)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 	at org.apache.james.mailbox.cassandra.mail.CassandraMessageIdMapperTest.provideMapper(CassandraMessageIdMapperTest.java:42)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] [ERROR] findShouldReturnMultipleMessagesWhenIdContainsMultiple  Time elapsed: 0.015 s  <<< ERROR!
[b5e8b12158a8a419f20041ff3cc374de36640fc9] com.google.inject.CreationException: 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] Unable to create injector, see the following errors:
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1) No implementation for org.apache.james.blob.api.BlobStore was bound.
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   while locating org.apache.james.blob.api.BlobStore
[b5e8b12158a8a419f20041ff3cc374de36640fc9]     for the 2nd parameter of org.apache.james.blob.api.DeduplicatingBlobStoreImpl.<init>(DeduplicatingBlobStoreImpl.java:43)
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   at org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils.lambda$commonModules$2(GuiceUtils.java:75) (via modules: com.google.inject.util.Modules$CombinedModule -> org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils$$Lambda$548/0x00000001005ff040)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1 error
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 	at org.apache.james.mailbox.cassandra.mail.CassandraMessageIdMapperTest.provideMapper(CassandraMessageIdMapperTest.java:50)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 	at org.apache.james.mailbox.cassandra.mail.CassandraMessageIdMapperTest.provideMapper(CassandraMessageIdMapperTest.java:42)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] [ERROR] deleteShouldDeleteMessageIndicesWhenStoredTwoTimesInTheSameMailbox  Time elapsed: 0.014 s  <<< ERROR!
[b5e8b12158a8a419f20041ff3cc374de36640fc9] com.google.inject.CreationException: 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] Unable to create injector, see the following errors:
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1) No implementation for org.apache.james.blob.api.BlobStore was bound.
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   while locating org.apache.james.blob.api.BlobStore
[b5e8b12158a8a419f20041ff3cc374de36640fc9]     for the 2nd parameter of org.apache.james.blob.api.DeduplicatingBlobStoreImpl.<init>(DeduplicatingBlobStoreImpl.java:43)
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   at org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils.lambda$commonModules$2(GuiceUtils.java:75) (via modules: com.google.inject.util.Modules$CombinedModule -> org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils$$Lambda$548/0x00000001005ff040)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1 error
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 	at org.apache.james.mailbox.cassandra.mail.CassandraMessageIdMapperTest.provideMapper(CassandraMessageIdMapperTest.java:50)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 	at org.apache.james.mailbox.cassandra.mail.CassandraMessageIdMapperTest.provideMapper(CassandraMessageIdMapperTest.java:42)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] [ERROR] findMailboxesShouldReturnTwoMailboxesWhenMessageExistsInTwoMailboxes  Time elapsed: 0.015 s  <<< ERROR!
[b5e8b12158a8a419f20041ff3cc374de36640fc9] com.google.inject.CreationException: 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] Unable to create injector, see the following errors:
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1) No implementation for org.apache.james.blob.api.BlobStore was bound.
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   while locating org.apache.james.blob.api.BlobStore
[b5e8b12158a8a419f20041ff3cc374de36640fc9]     for the 2nd parameter of org.apache.james.blob.api.DeduplicatingBlobStoreImpl.<init>(DeduplicatingBlobStoreImpl.java:43)
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   at org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils.lambda$commonModules$2(GuiceUtils.java:75) (via modules: com.google.inject.util.Modules$CombinedModule -> org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils$$Lambda$548/0x00000001005ff040)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1 error
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 	at org.apache.james.mailbox.cassandra.mail.CassandraMessageIdMapperTest.provideMapper(CassandraMessageIdMapperTest.java:50)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 	at org.apache.james.mailbox.cassandra.mail.CassandraMessageIdMapperTest.provideMapper(CassandraMessageIdMapperTest.java:42)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] [ERROR] findShouldReturnCorrectElementsWhenChunking  Time elapsed: 0.014 s  <<< ERROR!
[b5e8b12158a8a419f20041ff3cc374de36640fc9] com.google.inject.CreationException: 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] Unable to create injector, see the following errors:
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1) No implementation for org.apache.james.blob.api.BlobStore was bound.
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   while locating org.apache.james.blob.api.BlobStore
[b5e8b12158a8a419f20041ff3cc374de36640fc9]     for the 2nd parameter of org.apache.james.blob.api.DeduplicatingBlobStoreImpl.<init>(DeduplicatingBlobStoreImpl.java:43)
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   at org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils.lambda$commonModules$2(GuiceUtils.java:75) (via modules: com.google.inject.util.Modules$CombinedModule -> org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils$$Lambda$548/0x00000001005ff040)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1 error
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 	at org.apache.james.mailbox.cassandra.mail.CassandraMessageIdMapperTest.provideMapper(CassandraMessageIdMapperTest.java:50)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 	at org.apache.james.mailbox.cassandra.mail.CassandraMessageIdMapperTest.provideMapper(CassandraMessageIdMapperTest.java:42)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] [INFO] Running org.apache.james.mailbox.cassandra.mail.CassandraMessageWithAttachmentMapperTest
[b5e8b12158a8a419f20041ff3cc374de36640fc9] [ERROR] Tests run: 8, Failures: 0, Errors: 8, Skipped: 0, Time elapsed: 0.01 s <<< FAILURE! - in org.apache.james.mailbox.cassandra.mail.CassandraMessageWithAttachmentMapperTest
[b5e8b12158a8a419f20041ff3cc374de36640fc9] [ERROR] messagesRetrievedUsingFetchTypeBodyShouldHaveAttachmentsLoadedWhenOneAttachment  Time elapsed: 0 s  <<< ERROR!
[b5e8b12158a8a419f20041ff3cc374de36640fc9] com.google.inject.CreationException: 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] Unable to create injector, see the following errors:
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1) No implementation for org.apache.james.blob.api.BlobStore was bound.
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   while locating org.apache.james.blob.api.BlobStore
[b5e8b12158a8a419f20041ff3cc374de36640fc9]     for the 2nd parameter of org.apache.james.blob.api.DeduplicatingBlobStoreImpl.<init>(DeduplicatingBlobStoreImpl.java:43)
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   at org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils.lambda$commonModules$2(GuiceUtils.java:75) (via modules: com.google.inject.util.Modules$CombinedModule -> org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils$$Lambda$548/0x00000001005ff040)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1 error
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 	at org.apache.james.mailbox.cassandra.mail.CassandraMessageWithAttachmentMapperTest.createMapperProvider(CassandraMessageWithAttachmentMapperTest.java:33)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] [ERROR] messagesCanBeRetrievedInMailboxWithRangeTypeOne  Time elapsed: 0 s  <<< ERROR!
[b5e8b12158a8a419f20041ff3cc374de36640fc9] com.google.inject.CreationException: 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] Unable to create injector, see the following errors:
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1) No implementation for org.apache.james.blob.api.BlobStore was bound.
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   while locating org.apache.james.blob.api.BlobStore
[b5e8b12158a8a419f20041ff3cc374de36640fc9]     for the 2nd parameter of org.apache.james.blob.api.DeduplicatingBlobStoreImpl.<init>(DeduplicatingBlobStoreImpl.java:43)
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   at org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils.lambda$commonModules$2(GuiceUtils.java:75) (via modules: com.google.inject.util.Modules$CombinedModule -> org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils$$Lambda$548/0x00000001005ff040)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1 error
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 	at org.apache.james.mailbox.cassandra.mail.CassandraMessageWithAttachmentMapperTest.createMapperProvider(CassandraMessageWithAttachmentMapperTest.java:33)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] [ERROR] messagesRetrievedUsingFetchTypeHeadersShouldHaveAttachmentsEmptyWhenOneAttachment  Time elapsed: 0 s  <<< ERROR!
[b5e8b12158a8a419f20041ff3cc374de36640fc9] com.google.inject.CreationException: 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] Unable to create injector, see the following errors:
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1) No implementation for org.apache.james.blob.api.BlobStore was bound.
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   while locating org.apache.james.blob.api.BlobStore
[b5e8b12158a8a419f20041ff3cc374de36640fc9]     for the 2nd parameter of org.apache.james.blob.api.DeduplicatingBlobStoreImpl.<init>(DeduplicatingBlobStoreImpl.java:43)
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   at org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils.lambda$commonModules$2(GuiceUtils.java:75) (via modules: com.google.inject.util.Modules$CombinedModule -> org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils$$Lambda$548/0x00000001005ff040)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1 error
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 	at org.apache.james.mailbox.cassandra.mail.CassandraMessageWithAttachmentMapperTest.createMapperProvider(CassandraMessageWithAttachmentMapperTest.java:33)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] [ERROR] messagesRetrievedUsingFetchTypeMetadataShouldHaveAttachmentsEmptyWhenOneAttachment  Time elapsed: 0 s  <<< ERROR!
[b5e8b12158a8a419f20041ff3cc374de36640fc9] com.google.inject.CreationException: 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] Unable to create injector, see the following errors:
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1) No implementation for org.apache.james.blob.api.BlobStore was bound.
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   while locating org.apache.james.blob.api.BlobStore
[b5e8b12158a8a419f20041ff3cc374de36640fc9]     for the 2nd parameter of org.apache.james.blob.api.DeduplicatingBlobStoreImpl.<init>(DeduplicatingBlobStoreImpl.java:43)
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   at org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils.lambda$commonModules$2(GuiceUtils.java:75) (via modules: com.google.inject.util.Modules$CombinedModule -> org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils$$Lambda$548/0x00000001005ff040)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1 error
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 	at org.apache.james.mailbox.cassandra.mail.CassandraMessageWithAttachmentMapperTest.createMapperProvider(CassandraMessageWithAttachmentMapperTest.java:33)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] [ERROR] messagesRetrievedUsingFetchTypeBodyShouldHaveBodyDataLoaded  Time elapsed: 0 s  <<< ERROR!
[b5e8b12158a8a419f20041ff3cc374de36640fc9] com.google.inject.CreationException: 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] Unable to create injector, see the following errors:
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1) No implementation for org.apache.james.blob.api.BlobStore was bound.
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   while locating org.apache.james.blob.api.BlobStore
[b5e8b12158a8a419f20041ff3cc374de36640fc9]     for the 2nd parameter of org.apache.james.blob.api.DeduplicatingBlobStoreImpl.<init>(DeduplicatingBlobStoreImpl.java:43)
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   at org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils.lambda$commonModules$2(GuiceUtils.java:75) (via modules: com.google.inject.util.Modules$CombinedModule -> org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils$$Lambda$548/0x00000001005ff040)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1 error
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 	at org.apache.james.mailbox.cassandra.mail.CassandraMessageWithAttachmentMapperTest.createMapperProvider(CassandraMessageWithAttachmentMapperTest.java:33)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] [ERROR] messagesRetrievedUsingFetchTypeFullShouldHaveAttachmentsLoadedWhenTwoAttachments  Time elapsed: 0 s  <<< ERROR!
[b5e8b12158a8a419f20041ff3cc374de36640fc9] com.google.inject.CreationException: 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] Unable to create injector, see the following errors:
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1) No implementation for org.apache.james.blob.api.BlobStore was bound.
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   while locating org.apache.james.blob.api.BlobStore
[b5e8b12158a8a419f20041ff3cc374de36640fc9]     for the 2nd parameter of org.apache.james.blob.api.DeduplicatingBlobStoreImpl.<init>(DeduplicatingBlobStoreImpl.java:43)
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   at org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils.lambda$commonModules$2(GuiceUtils.java:75) (via modules: com.google.inject.util.Modules$CombinedModule -> org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils$$Lambda$548/0x00000001005ff040)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1 error
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 	at org.apache.james.mailbox.cassandra.mail.CassandraMessageWithAttachmentMapperTest.createMapperProvider(CassandraMessageWithAttachmentMapperTest.java:33)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] [ERROR] messagesRetrievedUsingFetchTypeFullShouldHaveAttachmentsLoadedWhenOneAttachment  Time elapsed: 0 s  <<< ERROR!
[b5e8b12158a8a419f20041ff3cc374de36640fc9] com.google.inject.CreationException: 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] Unable to create injector, see the following errors:
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1) No implementation for org.apache.james.blob.api.BlobStore was bound.
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   while locating org.apache.james.blob.api.BlobStore
[b5e8b12158a8a419f20041ff3cc374de36640fc9]     for the 2nd parameter of org.apache.james.blob.api.DeduplicatingBlobStoreImpl.<init>(DeduplicatingBlobStoreImpl.java:43)
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   at org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils.lambda$commonModules$2(GuiceUtils.java:75) (via modules: com.google.inject.util.Modules$CombinedModule -> org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils$$Lambda$548/0x00000001005ff040)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1 error
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 	at org.apache.james.mailbox.cassandra.mail.CassandraMessageWithAttachmentMapperTest.createMapperProvider(CassandraMessageWithAttachmentMapperTest.java:33)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] [ERROR] messagesRetrievedUsingFetchTypeFullShouldHaveAttachmentsEmptyWhenNoAttachment  Time elapsed: 0 s  <<< ERROR!
[b5e8b12158a8a419f20041ff3cc374de36640fc9] com.google.inject.CreationException: 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] Unable to create injector, see the following errors:
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1) No implementation for org.apache.james.blob.api.BlobStore was bound.
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   while locating org.apache.james.blob.api.BlobStore
[b5e8b12158a8a419f20041ff3cc374de36640fc9]     for the 2nd parameter of org.apache.james.blob.api.DeduplicatingBlobStoreImpl.<init>(DeduplicatingBlobStoreImpl.java:43)
[b5e8b12158a8a419f20041ff3cc374de36640fc9]   at org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils.lambda$commonModules$2(GuiceUtils.java:75) (via modules: com.google.inject.util.Modules$CombinedModule -> org.apache.james.mailbox.cassandra.mail.utils.GuiceUtils$$Lambda$548/0x00000001005ff040)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 1 error
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 	at org.apache.james.mailbox.cassandra.mail.CassandraMessageWithAttachmentMapperTest.createMapperProvider(CassandraMessageWithAttachmentMapperTest.java:33)
[b5e8b12158a8a419f20041ff3cc374de36640fc9] 
[b5e8b12158a8a419f20041ff3cc374de36640fc9] [INFO] Running org.apache.james.mailbox.cassandra.mail.CassandraMailboxDAOTest
[b5e8b12158a8a419f20041ff3cc374de36640fc9] [INFO] Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.296 s - in org.apache.james.mailbox.cassandra.mail.CassandraMailboxDAOTest
[b5e8b12158a8a419f20041ff3cc374de36640fc9] [INFO] Running org.apache.james.mailbox.cassandra.mail.CassandraMessageIdDAOTest
[b5e8b12158a8a419f20041ff3cc374de36640fc9] [INFO] Tests run: 18, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.601 s - in org.apache.james.mailbox.cassandra.mail.CassandraMessageIdDAOTest
[b5e8b12158a8a419f20041ff3cc374de36640fc9] [INFO] Running org.apache.james.mailbox.cassandra.ids.CassandraMessageIdTest
[b5e8b12158a8a419f20041ff3cc374de36640fc9] [INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.027 s - in org.apache.james.mailbox.cassandra.ids.CassandraMessageIdTest
[b5e8b12158a8a419f20041ff3cc374de36640fc9] [INFO] Running org.apache.james.mailbox.cassandra.ids.CassandraMailboxIdDeserializerTest
[b5e8b12158a8a419f20041ff3cc374de36640fc9] [INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.003 s - in 
```

File: server/blob/blob-api/pom.xml (hunk 1 above)
Comment (chibenwa) [reply server/blob/blob-api/pom.xml]: Is it ok to have Impl in the same maven module than APIs ?

That's good to me but I want to ask it out loud...

Comment (remk) [reply server/blob/blob-api/pom.xml]: It's ok for me, let's ask the more opinionated team members @mbaechler ? :smile:

**Comment by remk** [comment]
At the time being the object storage blob store separation between low level blob store and the high level one is not done. So the DeletedMessage vault can use only the cassandra or memory low level blobstore. We need to rebase on the PR which do this separation for jcloud or s3.

File: mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/blob/BlobStoreDeletedMessageVault.java [hunk 3]
```diff
@@ -66,21 +65,24 @@
 
     private final MetricFactory metricFactory;
     private final DeletedMessageMetadataVault messageMetadataVault;
-    private final DeduplicatingBlobStore blobStore;
+    private final BlobStore blobStore;
```
Comment (chibenwa) [inline mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/blob/BlobStoreDeletedMessageVault.java]: I think deduplication is expected 

Cf `MemoryDeletedMessageVaultIntegrationTest>DeletedMessageVaultIntegrationTest.vaultDeleteShouldDeleteAllMessagesHavingSameBlobContent`

That's what I had been told when I came up with https://github.com/linagora/james-project/pull/3028

Could you at the very least move this commit in a separated pull request?

**Comment by chibenwa** [comment]
Not relevant anymore

**Pull request closed without merging**